.github/workflows/test-and-push-docker-image.yaml

```diff
@@ -0,0 +1,122 @@
+name: Run tests and push Docker image on success
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+
+  test-js:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [8.x, 10.x, 12.x]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
```

Comment:
Should we cache using the matrix version too? Given the packages are likely to be different between Node versions.

Or should we just pick one version of Node

Reply:
> Should we cache using the matrix version too? Given the packages are likely to be different between Node versions.

I thought about that, and my conclusion was that it wouldn't matter and `npm install` would handle that anyway. So I kept it simple. TBH, this caching doesn't make much of a difference in our case as run the `npm install` took between 16-20 seconds before the caching 😬 

What do you reckon?

> Or should we just pick one version of Node
We weren't doing that in the old CircleCI workflow but I thought it was a good idea as it will give us visibility over JS stuff breaking in future node versions before we decide to upgrade.

Reply:
From https://github.com/actions/cache/blob/main/examples.md#node---npm : 

> Note: It is not recommended to cache node_modules, as it can break across Node versions and won't work with npm ci

It doesn't seem to break in our case but I'll tell you what, since the benefit is marginal (maybe 10s in the best case) I'll just keep it simple and remove the caching of this altogether unless you have any other insight here.